src/mon/OSDMonitor.cc

```diff
@@ -3261,6 +3261,12 @@ epoch_t OSDMonitor::send_pg_creates(int osd, Connection *con, epoch_t next) cons
   dout(30) << __func__ << " osd." << osd << " next=" << next
 	   << " " << creating_pgs_by_osd_epoch << dendl;
   std::lock_guard<std::mutex> l(creating_pgs_lock);
+  if (creating_pgs_epoch < mapping.get_epoch()) {
+    dout(20) << __func__
+	     << " not using stale creating_pgs@" << creating_pgs_epoch << dendl;
+    // the subscribers will be updated when the mapping is completed anyway
+    return next;
+  }
```

Comment:
@gregsfortytwo `C_UpdateCreatingPGs` calls `update_creating_pgs()` and `check_pg_creates_subs()` when it's done with the mapping. so i think, once `creating_pgs_by_osd_epoch` is sync'ed with `creating_pgs.pgs`, we will send the pg creates to OSDs.

and we always try to start the mapping when the osdmap is committed, see `OSDMonitor::on_active()`. so it's safe to assume so, i believe.